src/openforms/submissions/models/submission.py

```diff
@@ -702,13 +702,24 @@ def calculate_price(self, save=True) -> None:
         """
         Calculate and save the price of this particular submission.
         """
+        logger.debug("Calculating submission %s price", self.uuid)
+        if not self.payment_required:
+            logger.debug(
+                "Submission %s does not require payment, skipping price calculation",
+                self.uuid,
+            )
+            return None
+
         self.price = get_submission_price(self)
         if save:
             self.save(update_fields=["price"])
 
     @property
     def payment_required(self) -> bool:
-        return self.form.payment_required
+        if not self.form.payment_required:
+            return False
+
+        return bool(get_submission_price(self))
```

Comment:
`get_submission_price` calls `submission.payment_required`, so this probably end up in an infinite loop?

Reply:
No, `calculate_price` is calling `submission.payment_required` according to the last commit. The `get_submission_price` is now only evaluating the rules or the product price

Reply:
I prefer this implementation using the `get_submission_price` call so that property access does not perform side effects anymore now.

I have some performance concerns, but instead of prematurely optimizing, let's keep the code as-is in this PR and keep an eye on APM for possible performance issues.

Reply:
Ah yes missed the diff on that part